source/MaterialXRuntime/Private/PrvTypeDef.cpp

```diff
@@ -0,0 +1,369 @@
+//
+// TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRuntime/Private/PrvTypeDef.h>
+
+#include <MaterialXRuntime/RtTypeDef.h>
+
+#include <MaterialXCore/Util.h>
+
+#include <sstream>
+
+namespace MaterialX
+{
+
+namespace
+{
+
+template<class T>
+RtValue createValue(RtObject&)
+{
+    return RtValue((T)0);
+}
+template<> RtValue createValue<Matrix33>(RtObject& owner)
+{
+    return RtValue(Matrix33::IDENTITY, owner);
+}
+template<> RtValue createValue<Matrix44>(RtObject& owner)
+{
+    return RtValue(Matrix44::IDENTITY, owner);
+}
+template<> RtValue createValue<string>(RtObject& owner)
+{
+    return RtValue(string(""), owner);
+}
+template<> RtValue createValue<RtToken>(RtObject&)
+{
+    return RtValue(EMPTY_TOKEN);
+}
+RtValue createNoneValue(RtObject&)
+{
+    return RtValue(0);
+}
+
+template<class T>
+void copyValue(const RtValue& src, RtValue& dest)
+{
+    dest = src;
+}
+template <> void copyValue<Matrix33>(const RtValue& src, RtValue& dest)
+{
+    dest.asMatrix33() = src.asMatrix33();
+}
+template <> void copyValue<Matrix44>(const RtValue& src, RtValue& dest)
+{
+    dest.asMatrix44() = src.asMatrix44();
+}
+template <> void copyValue<string>(const RtValue& src, RtValue& dest)
+{
+    dest.asString() = src.asString();
+}
+void copyNoneValue(const RtValue&, RtValue&)
+{
+}
+
+template<class T>
+void marshalValue(const RtValue&, string&)
```

Comment:
I'm not sure if we want to expose these as computer terms of marhalling and unmarshalling. Maybe something like toString() and fromString() instead since it gives you actual target data type.

Reply:
Good point, I will rename these.